modules/promotion/tests/src/Kernel/UsageTest.php

```diff
@@ -0,0 +1,217 @@
+
+use Drupal\commerce_order\Entity\Order;
+use Drupal\commerce_order\Entity\OrderInterface;
+use Drupal\commerce_order\Entity\OrderItemType;
+use Drupal\commerce_promotion\Entity\Coupon;
+use Drupal\commerce_promotion\Entity\CouponInterface;
+use Drupal\commerce_promotion\Entity\Promotion;
+use Drupal\commerce_promotion\Entity\PromotionInterface;
+use Drupal\Tests\commerce\Kernel\CommerceKernelTestBase;
+use Drupal\user\UserInterface;
+
+/**
+ * Tests the usage tracking of promotions.
+ *
+ * @group commerce
+ * @group commerce_promotion
+ *
+ * @coversDefaultClass \Drupal\commerce_promotion\PromotionUsage
+ */
+class UsageTest extends CommerceKernelTestBase {
+
+  /**
+   * The coupon storage.
+   *
+   * @var \Drupal\commerce_promotion\CouponStorageInterface
+   */
+  protected $couponStorage;
+
+
+  /**
+   * The promotion storage.
+   *
+   * @var \Drupal\commerce_promotion\PromotionStorageInterface
+   */
+  protected $promotionStorage;
+
+  /**
+   * The usage.
+   *
+   * @var \Drupal\commerce_promotion\PromotionUsageInterface
+   */
+  protected $usage;
+
+  /**
+   * The test order.
+   *
+   * @var \Drupal\commerce_order\Entity\OrderInterface
+   */
+  protected $order;
+
+  /**
+   * Modules to enable.
+   *
+   * @var array
+   */
+  public static $modules = [
+    'entity_reference_revisions',
+    'profile',
+    'state_machine',
+    'commerce_order',
+    'commerce_product',
+    'commerce_promotion',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    $this->installEntitySchema('profile');
+    $this->installEntitySchema('commerce_order');
+    $this->installEntitySchema('commerce_order_type');
+    $this->installEntitySchema('commerce_promotion');
+    $this->installEntitySchema('commerce_promotion_coupon');
+    $this->installConfig([
+      'profile',
+      'commerce_order',
+      'commerce_promotion',
+    ]);
+    $this->installSchema('commerce_promotion', ['commerce_promotion_usage']);
+
+    $this->couponStorage = $this->container->get('entity_type.manager')->getStorage('commerce_promotion_coupon');
+    $this->promotionStorage = $this->container->get('entity_type.manager')->getStorage('commerce_promotion');
+    $this->usage = $this->container->get('commerce_promotion.usage');
+
+    $this->user = $this->createUser();
+
+    OrderItemType::create([
+      'id' => 'test',
+      'label' => 'Test',
+      'orderType' => 'default',
+    ])->save();
+
+    $this->order = Order::create([
+      'type' => 'default',
+      'state' => 'draft',
+      'mail' => 'test@example.com',
+      'ip_address' => '127.0.0.1',
+      'order_number' => '6',
+      'store_id' => $this->store,
+      'uid' => $this->user,
+      'order_items' => [],
+    ]);
+  }
+
+  /**
+   * Manually test the add/get of usages.
+   *
+   * @covers ::getPromotionUsage
+   * @covers ::addPromotionUsage
+   * @covers ::getCouponUsage
+   * @covers ::addCouponUsage
+   */
+  public function testMockUsage() {
```

Comment:
@bojanz this method at leasts tests the logic and storage. Order refresh and legitimate implementation will take some time. If you'd like to review the table name, columns, and basic storage logic.